apps/client/src/common/utils/aliases.ts

```diff
@@ -26,4 +29,42 @@ export const validateAlias = (alias: string) => {
   }
 
   return valid;
-};
\ No newline at end of file
+};
+
+export const getAliasRoute = (location: Location, data: Alias[], searchParams: URLSearchParams) => {
+  const currentURL = location.pathname.substring(1);
+  let redirectURL = '';
```

Comment:
I think the code could be slightly easier to read if we don't keep this around since we have early returns
so the code would look like

```
if (foundAlias) {
   return generateURLFromAlias(foundAlias);
}

for loop {
    if aliasMatch {
        return `${newAliasPath.pathname}?${urlParams.toString()}`;
   }
}

return null

```'

This means less indentations and clauses to keep track of